src/psyclone/domain/lfric/kernel/field_vector_arg.py

```diff
@@ -89,14 +91,21 @@ def create_from_fparser2(fparser2_tree):
         :returns: an instance of this class.
         :rtype: :py:class:`psyclone.domain.lfric.kernel.FieldVectorArg`
 
-        :raises TypeError: if the first metadata argument is not in \
-            the form datatype*vector_length.
+        raises ValueError: if the metadata is not in the correct form.
 
         '''
         FieldVectorArg.check_fparser2(fparser2_tree, nargs=4)
-        vector_length = FieldVectorArg.get_vector_length(fparser2_tree)
+        FieldVectorArg.check_first_arg(
+            fparser2_tree, "FieldVector", vector=True)
+        vector_length = FieldVectorArg.get_and_check_vector_length(
+            fparser2_tree, FieldVectorArg.vector_length_arg_index)
         datatype, access, function_space = \
-            FieldVectorArg.get_type_access_and_fs(fparser2_tree)
+            FieldVectorArg.get_type_access_and_fs(
```

Comment:
If you made `get_type_access_and_fs` an instance (?) method rather than a class method, would you be able to avoid having to pass in the `access_arg_index` and `function_space_arg_index` values?

Reply:
Made into a class method as you suggest in your next comment. This makes it much nicer, thanks.